test/functional/change_stream.test.js

```diff
@@ -61,7 +66,7 @@ function triggerResumableError(changeStream, delay, onClose) {
   });
 
   function triggerError() {
-    changeStream.cursor.emit('error', new MongoNetworkError('fake error'));
+    changeStream.triggerError(new MongoNetworkError('fake error'));
```

Comment:
rather than creating a special method (`triggerError`) for behavior we wish to test, what if you made an internal accessor for the stream? 
```
class ChangeStream { 
  /** @internal */
  get cursorStream() {
    return this[kCursorStream];
  }
}
```

If you had that, then you would be able to renable the two tests below as well

Reply:
Unfortunately that approach only works when there *is* an internal stream we can emit events on. When the change stream is being iterated, there's no internal stream and therefore nowhere to inject the error without something that exposes `processError` more directly like this.